src/queue.js

```diff
@@ -219,4 +219,6 @@ class Queue extends EventEmitter {
   }
 }
 
+Queue.Job = null; // Make TypeScript compiler happy. Needed because Job is an abstract class.
```

Comment:
Well, those are good questions.

As far as using 'null' is concerned, I kind of discovered this by accident - the error messages from the compiler complained that the base class had to be either a function or null. Since I knew that you had objected to exposing Job in the earlier PR, I thought this might be better. As to whether it's the right choice, the compiler accepts it, and we don't want user code to be able to call the Job constructor function (or directly construct any subclass of Job) so it seems OK not to export the actual function value.

As to the issue of naming, here's some more background: In TypeScript, every name can have up to three possible definitions: as a value, as a type, and as a namespace. So for example, it's possible to have both a value named Queue (the constructor function) and a namespace named Queue defined in the same source file. Note that classes define both a value and a type (but not a namespace).

What's even more mind-bending is that when you import the name 'Queue' it actually imports all three of those meanings. I know this seems odd, but this appears to be intended to solve the rather thorny problems modeling CommonJS exports in a statically typed system.

Since rethinkdb-job-queue only exports one symbol, Queue, using CommonJS default export, the question is how to export other symbols such as Job. The ES6 syntax:

    import Queue, { Job } from 'queue';

...won't work with CommonJS default exports. Instead you have to use:

    import * as Queue from 'queue';

So the only other option is to declare a namespace Queue (same name as the exported function) and put Job and all the other types inside of it. And if those types were only interfaces, that would be the end of the story; But because of my decision to make Job an abstract class, something extra is required - there needs to be a value somewhere.